src/contexts/global.tsx

```diff
@@ -228,6 +230,25 @@ export const GlobalContextProvider: FC<{ overrideContextValue?: PartialDeep<Glob
     }
   }, [currentNetwork, networkStatus, t])
 
+  useEffect(
+    () => {
+      _window.electron?.changeTheme(settings.general.theme)
+      const noop = () => {
+        /* do nothing */
+      }
+      const removeListeners: (() => void)[] = []
+      removeListeners.push(
+        _window.electron?.onUpdateThemeDark(() => updateSettings('general', { theme: 'dark' })) || noop
+      )
+      removeListeners.push(
+        _window.electron?.onUpdateThemeLight(() => updateSettings('general', { theme: 'light' })) || noop
+      )
+      return () => removeListeners.forEach((r) => r())
```

Comment:
Instead of having this noop/doNothing function, wouldn't it work the same if we just check first if `_window.electron` is defined and in this case push the function call to the listeners array?

```ts
  useEffect(() => {
    const removeListeners: (() => void)[] = []
    const electron = _window.electron

    if (electron) {
      electron.changeTheme(settings.general.theme)

      removeListeners.push(electron.onUpdateThemeDark(() => updateSettings('general', { theme: 'dark' })))
      removeListeners.push(electron.onUpdateThemeLight(() => updateSettings('general', { theme: 'light' })))
    }

    return () => removeListeners.forEach((r) => r())
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [])
```

Reply:
Yeah, looks good :slightly_smiling_face: